radicle-ssh/src/agent/client.rs

```diff
@@ -297,70 +293,71 @@ impl<S: ClientStream> AgentClient<S> {
 
         let total = 1 + pk.len();
 
-        self.buf.zeroize();
-        self.buf.write_u32::<BigEndian>(total as u32)?;
-        self.buf.push(msg::REMOVE_IDENTITY);
-        self.buf.extend_from_slice(&pk);
+        let mut buf = Buffer::default();
+        buf.write_u32::<BigEndian>(total as u32)?;
+        buf.push(msg::REMOVE_IDENTITY);
+        buf.extend_from_slice(&pk);
 
-        self.stream.read_response(&mut self.buf)?;
+        self.stream.request(&buf)?;
 
         Ok(())
     }
 
     /// Ask the agent to remove a smartcard from its memory.
     pub fn remove_smartcard_key(&mut self, id: &str, pin: &[u8]) -> Result<(), Error> {
-        self.buf.zeroize();
-        self.buf.resize(4, 0);
-        self.buf.push(msg::REMOVE_SMARTCARD_KEY);
-        self.buf.extend_ssh_string(id.as_bytes());
-        self.buf.extend_ssh_string(pin);
-        self.buf.write_len();
+        let mut buf = Buffer::default();
+        buf.resize(4, 0);
+        buf.push(msg::REMOVE_SMARTCARD_KEY);
+        buf.extend_ssh_string(id.as_bytes());
+        buf.extend_ssh_string(pin);
+        buf.write_len();
 
-        self.stream.read_response(&mut self.buf)?;
+        self.stream.request(&buf)?;
 
         Ok(())
     }
 
     /// Ask the agent to forget all known keys.
     pub fn remove_all_identities(&mut self) -> Result<(), Error> {
-        self.buf.zeroize();
```

Comment:
Zeroizing has been removed without explanation.  Is there a security concern around it?

Reply:
Check line 322, we create a new zeroizing buffer for each request, so there's no need to zeroize manually.